dds/idl/marshal_generator.cpp

```diff
@@ -3021,10 +3035,33 @@ bool marshal_generator::gen_union(AST_Union* node, UTL_ScopedName* name,
     code.push_back("}");
     generate_dheader_code(code, not_final);
 
+    // Header for discriminator
```

Comment:
If this is just for XCDR2, I think it would be more clear if we called this EMHEADER to distinguish it from other kinds of headers.